packages/dustkit/src/apps/getProgramAppConfigUrl.ts

```diff
@@ -18,17 +18,30 @@ export async function getProgramAppConfigUrl({
   program: ProgramId;
   entity?: EntityId;
 }): Promise<string | undefined> {
-  const url = await readContract(client, {
+  let url: string | undefined;
+  const { system } = await getRecord(client, {
     address: worldAddress,
-    abi: worldCallAbi,
-    functionName: "call",
-    args: encodeSystemCall({
+    table: baseWorldConfig.tables.world__Systems,
+    key: {
       systemId: program,
+    },
+  });
+
+  try {
+    // We are calling the program system directly here, as programs are required to be private
+    // systems so that their hook functions can only be called by the world
+    url = await readContract(client, {
+      address: system,
       abi: appAbi,
       functionName: "appConfigURI",
       args: [entity ?? zeroHash],
-    }),
-  });
+    });
+  } catch (error: any) {
+    if (error?.name === "ContractFunctionExecutionError") {
```

Comment:
can we be any more specific than this?

Reply:
<img width="490" alt="image" src="https://github.com/user-attachments/assets/b6ffa3cc-fb24-4f0f-914e-595da1baa4fa" />

Reply:
This is the error I get when theres no method

![image](https://github.com/user-attachments/assets/c63594d8-5fed-41b9-965f-a1261dbcd40a)